### PR TITLE
Rework line break handling

### PR DESCRIPTION
**Pull request by Flamefire**
To fix https://github.com/Return-To-The-Roots/s25client/issues/977 this adds the U+FFFD glyph to show unprintable chars and treats newlines in fonts as such. So all line breaks must be handled outside of `font.Draw()` e.g. via `ctrlMultiline`

Also the font class has been refactored: Only UTF8 text is handled now (other text (mostly) wasn't used), which needed changes in ctrlEdit. I added tests for that too. Also the Font class now is constructed from a Font-ArchivItem instead of being one. This initializes the class eagerly (on ctor) instead of lazily allowing use as const (why would a `Draw` be non-const?)

Especially the line-break-handling removal brought a ~5% performance gain when drawing text. I expect the new class to perform generally better due to smaller memory usage and less checks to be done as well as being non-virtual avoiding a `dynamic_cast` in the loader

For the future I'd model all classes like that: Separate ArchivItem and actual usage. E.g. we don't need 6 different `gl*Bitmap` classes, but only 1 (or 2 for player) sprite or texture class. So during loading the classes are "transformed"/loaded into GPU or whatever and then used like that. Cut down on `dynamic_cast`s at runtime.

### CODE REVIEW

File: libs/s25main/Loader.cpp [hunk 1]
```diff
@@ -264,6 +269,25 @@ bool Loader::LoadSounds()
     return true;
 }
 
+bool Loader::LoadFonts()
+{
+    if(!LoadFile(RTTRCONFIG.ExpandPath("<RTTR_RTTR>/LSTS/OUTLINE_FONTS.LST"), GetPaletteN("pal5")))
+        return false;
+    fonts.clear();
+    const auto& loadedFonts = GetArchive("outline_fonts");
+    for(int i = 0; i < 3; i++)
```
Comment (Flow86) [inline libs/s25main/Loader.cpp]: we should make sure that the archive contains 3 fonts here

Comment (Flamefire) [reply libs/s25main/Loader.cpp]: It does: See 3 lines below

Comment (Flow86) [reply libs/s25main/Loader.cpp]: ah okay. I thought about a good way to not hardcode the "3" here - perhaps at least add some kind of constant `FONT_COUNT` or similar

File: libs/s25main/Loader.cpp [hunk 2]
```diff
@@ -303,13 +327,11 @@ void Loader::LoadDummyGUIFiles()
         io.push(std::move(bmp));
     }
     // Fonts
-    libsiedler2::Archiv& fonts = files_["outline_fonts"].archiv;
     auto* palette = GetPaletteN("colors");
-    fonts.alloc(3);
     libsiedler2::PixelBufferBGRA buffer(15, 16);
     for(unsigned i = 0; i < 3; i++)
```
Comment (Flow86) [inline libs/s25main/Loader.cpp]: same point is here with the "3"

File: libs/s25main/Loader.cpp (hunk 1 above)
Comment (Flamefire) [reply libs/s25main/Loader.cpp]: I added an enum `FontSize` and changed the signature of `GetFont` accordingly. This also makes sure you can only get valid fonts.

**Pull request merged**